build: support filtering cache by duration using `--cache-ttl` (#4165)
containers/buildah: docs/buildah-build.1.md

Note: `--cache-to` option is ignored unless `--layers` is specified.

> **--cache-ttl**

Reviewer: ```suggestion
**--cache-ttl** *duration*
```

> Limit use of cached images to only consider images with create timestamps less than *duration* ago.

Reviewer 2: I think you meant:
```suggestion
Limit the use of cached images to only consider images with created timestamps less than *duration* ago.
```

> For example if `--cache-ttl=1h` is specified buildah will only consider intermediate cache images which are created

Reviewer 2: ```suggestion
For example if `--cache-ttl=1h` is specified, Buildah will only consider intermediate cache images which are created
```

> under the duration of `1h` and intermediate cache images outside this duration will be ignored.

Reviewer 3: duration of one hour ...

> Accepts time duration and will only consider cached intermediate images created under this specified duration.
> For example if `--cache-ttl=1h` is specified buildah will only consider intermediate cache images which are created
> under the duration of one hour and intermediate cache images outside this duration will be ignored.

Reviewer: ```suggestion
Limit use of cached images to only consider images with create timestamps less than *duration* ago.
```
"create timestamp" is the phrasing we use to describe the "--timestamp" option, which someone is sure to try to use in combination with this one at some point.

Author: Amended but left the example line just in case someone wants to look at the example but can drop if needed.

> under the duration of one hour and intermediate cache images outside this duration will be ignored.

Reviewer 2: ```suggestion
under the duration of one hour, and intermediate cache images outside this duration will be ignored.
```
